File: Interpreted/lcs-recursive/lcs.lua
function lcs(xstr, ystr)
    if (xstr == "" or ystr == "") then return "" end;

    local x = xstr:sub(0,1);
    local xs = xstr:sub(2,-1);
    local y = ystr:sub(0,1);
    local ys = ystr:sub(2,-1);

    if x == y then
        return (x .. lcs(xs, ys))
    end

    local lcs1 = lcs(xstr, ys);
    local lcs2 = lcs(xs, ystr);

    if lcs1:len() > lcs2:len() then
        return lcs1
    else
        return lcs2
    end
end

print(lcs(arg[1], arg[2]))

File: Interpreted/lcs-recursive/lcs.py
import sys

def lcs(xstr, ystr):

    if (xstr == "" or ystr == ""):
        return "";

    x = xstr[0];
    xs = xstr[1:];
    y = ystr[0];
    ys = ystr[1:];

    if x == y:
        return(x + lcs(xs, ys));

    lcs1 = lcs(xstr, ys);
    lcs2 = lcs(xs, ystr);

    if len(lcs1) > len(lcs2):
        return lcs1
    else:
        return lcs2;


print(lcs(sys.argv[1], sys.argv[2]));
